app/containers/DatasetPage.js

```diff
@@ -124,11 +179,67 @@ class DatasetPage extends Component {
       }
     )
   }
+  handleUpload = (e) => {
+    const files = e.target.files
+    const f = files[0]
+    const datasetInfo = {
+      id: this.props.dataset.get('id'),
+      name: this.props.dataset.get('name'),
+      configuration: {
+        format: 'upload',
+        options: {
+          base: f.name,
+        }
+      }
+    }
+    this.state.uploadFileName = f.name
+    const reader = new FileReader();
+    reader.onload = ((e) => {
+      try {
+        const datasetConfig = parse(e.target.result)
+        this.props.importDataConfig(datasetInfo, datasetConfig)
+        this.props.updateDataset(datasetInfo)
```

Comment:
It is duplicated. 
I used `unpdateDataset` previously because we have an action `DATASET_SETTINGS_UPDATE`, whose next step is `DATASET_CONFIG_SAVE`. This is actually the action I was looking for. I needed to get the read-from-yml dataconfig saved. 
By that time I misunderstood the code and was confused about how to dispatch it directly. 
The problem is fixed now. The method `updateDataset` is removed, and I export a new function in `action/dataset.js` called `datasetSaveAll` that is able to directly dispatch the `DATASET_CONFIG_SAVE` action.